File: example/builtin/test_env.js
const xuanwu = require('xuanwu');

/**
 * 内建工具环境变量管理测试示例 (Node.js)
 * 
 * 使用说明：
 * 1. 确保已在该 Node.js 环境下安装过内建包。
 * 2. 运行时系统需要注入有效的 OpenAPI 凭证 (XWPKG_OPENAPI_TOKEN 或 OPENAPI_TOKEN)。
 */

async function main() {
    console.log("====== 开始运行 Node.js 环境变量管理示例 ======");

    try {
        // 获取全部环境变量
        const envs = await xuanwu.getEnvs();
        console.log(`当前共有 ${envs.length} 个环境变量`);

        // 新增一个临时环境变量
        const newEnvName = "XWPKG_TEST_KEY_JS";
        const newEnvVal = "HelloXuanwuJS";
        console.log(`正在创建环境变量: ${newEnvName}...`);
        const createdEnv = await xuanwu.addEnv(
            newEnvName,
            newEnvVal,
            "Node.js SDK 测试自动创建"
        );
        console.log(`创建成功: ID=${createdEnv.id}, Name=${createdEnv.name}`);

        // 查询该环境变量
        const checkedEnv = await xuanwu.getEnv(newEnvName);
        if (checkedEnv) {
            console.log(`成功查询到变量: ${checkedEnv.name} = ${checkedEnv.value}`);

            // 修改该环境变量的值
            const updatedVal = "HelloXuanwuJS_Updated";
            console.log(`正在修改环境变量的值为: ${updatedVal}...`);
            const updatedEnv = await xuanwu.updateEnv(
                checkedEnv.id,
                newEnvName,
                updatedVal,
                "Node.js SDK 测试自动更新"
            );
            console.log(`更新成功: Value=${updatedEnv.value}`);

            // 删除该临时环境变量
            console.log(`正在删除临时环境变量: ID=${checkedEnv.id}...`);
            await xuanwu.deleteEnv(checkedEnv.id);
            console.log("删除成功！");
        }

    } catch (e) {
        console.error(`环境变量操作失败: ${e.message}`);
        console.log("提示: 请确保在面板任务设置中正确注入了 OpenAPI Token。");
    }
}

main();

File: example/builtin/test_env.py
import xuanwu

# 内建工具环境变量管理测试示例 (Python)
# 
# 前提条件：
# 1. 已经在环境中安装了 xuanwu 包（例如通过 `xuanwu builtininstall`）
# 2. 环境中已注入有效环境变量：
#    - XWPKG_OPENAPI_TOKEN (必填，用于管理接口鉴权)

def main():
    print("====== 开始运行 Python 环境变量管理示例 ======")

    try:
        # 获取全部环境变量
        envs = xuanwu.get_envs()
        print(f"当前共有 {len(envs)} 个环境变量")
        
        # 新增一个临时环境变量
        new_env_name = "XWPKG_TEST_KEY"
        new_env_val = "HelloXuanwu"
        print(f"正在创建环境变量: {new_env_name}...")
        created_env = xuanwu.add_env(
            name=new_env_name,
            value=new_env_val,
            remark="Python SDK 测试自动创建"
        )
        print(f"创建成功: ID={created_env.get('id')}, Name={created_env.get('name')}")

        # 查询刚才创建的环境变量详情
        checked_env = xuanwu.get_env(new_env_name)
        if checked_env:
            print(f"成功查询到变量: {checked_env.get('name')} = {checked_env.get('value')}")

            # 修改该环境变量的值
            updated_val = "HelloXuanwu_Updated"
            print(f"正在修改环境变量的值为: {updated_val}...")
            updated_env = xuanwu.update_env(
                id=checked_env.get("id"),
                name=new_env_name,
                value=updated_val,
                remark="Python SDK 测试自动更新"
            )
            print(f"更新成功: Value={updated_env.get('value')}")

            # 删除该临时环境变量
            print(f"正在删除临时环境变量: ID={checked_env.get('id')}...")
            xuanwu.delete_env(checked_env.get("id"))
            print("删除成功！")
            
    except Exception as e:
        print(f"环境变量操作失败: {e}")
        print("提示: 请确保在面板任务设置中正确注入了 OpenAPI Token。")

if __name__ == "__main__":
    main()
